crates/nu-command/src/network/url/parse.rs

```diff
@@ -0,0 +1,184 @@
+use super::url;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Value,
+};
+
+use url::Url;
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "url parse"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("url parse")
+            .rest(
+                "rest",
+                SyntaxShape::CellPath,
+                "optionally operate by cell path",
+            )
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Parses a url"
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["url", "parse"]
```

Comment:
I believe that we don't add any words from the command name itself to the search terms: just any related terms that you believe users might enter when looking for this.

Reply:
hmm thanks for the explanation, so maybe "scheme", "hostname", "port", "path", and "parameters" ? That it's what the command returns